### PR TITLE
#309 update icons component DO NOT MERGE (needs work)

### PR DESCRIPTION
**Pull request by remydenton**
Please update checkboxes on #309 when merging

This is an attempt to update the icon and icons components.  It has the following issues/concerns:

1. Currently the build fails with this error:
`Module not found: Can't resolve 'fs' in '/Users/dentr1/Sites/bolt/node_modules/fs.realpath'`

2. The previous icon package had the following lines in package.json.  Do we have support for these in the new setup?
```
  "main": "dist/bolt-icon.min.js",
  "main:wwwd8": "dist/wwwd8/icon.min.js",
  "module:dev": "src/icon.js",
```

3. Similarly, the previous icons package had the following lines that may or may not make sense to keep:
```
  "main": "dist/index.js",
  "module": "dist/index.es6.js",
  "jsnext:main": "dist/index.es6.js",
```

### CODE REVIEW

File: packages/components/bolt-icons/package.json [hunk 1]
```diff
@@ -37,13 +36,5 @@
...
     "svgo": "^0.7.2",
     "uppercamelcase": "^3.0.0"
-  },
-  "main": "dist/index.js",
-  "module": "dist/index.es6.js",
-  "jsnext:main": "dist/index.es6.js",
-  "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
-    "prepublishOnly": "npm run build",
-    "build": "node ./build-icons.js && babel src --out-dir dist --ignore src/svgs"
```
Comment (sghoweri) [inline packages/components/bolt-icons/package.json]: @remydenton @EvanLovely this is the Icons-specific build script I mentioned earlier that I'd like to see us either port to Webpack and/or add as a new build task (likely the latter at a minimum given the processing going on)

Comment (sghoweri) [reply packages/components/bolt-icons/package.json]: ^ ideally we roll this into a standalone Webpack plugin -- off the shelf or modified

File: packages/components/bolt-icons/package.json [hunk 2]
```diff
@@ -37,13 +36,5 @@
     "prettier": "^1.7.4",
     "svgo": "^0.7.2",
     "uppercamelcase": "^3.0.0"
-  },
-  "main": "dist/index.js",
```
Comment (sghoweri) [inline packages/components/bolt-icons/package.json]: @remydenton to answer your question on whether or not we should remove these: it depends on where the build process for this lands. If we don't port this build task over now, we need to leave these in for now and pre-compile these prior to Webpack running.

Granted, that's not ideal however it did get us this for when we've been in crunch mode.

With regards to the "module" and jsnext" fields - that's part of my grand vision of getting us to eventually support native ES modules for supporting browsers (via `<script type="module">`) while also providing a transpiled version using normal script tags.

We're good to remove these.... for now 😉

File: packages/components/bolt-icons/package.json [hunk 3]
```diff
@@ -24,6 +22,7 @@
   "publishConfig": {
     "access": "public"
   },
+  "main": "build-icons.js",
```
Comment (sghoweri) [inline packages/components/bolt-icons/package.json]: @remydenton not the entry point JS file you're looking for 😉

**Comment by remydenton** [comment]
Closing in favor of https://github.com/bolt-design-system/bolt/pull/383/commits/84dc35cac536dff93dd39db21371c4a59c723218

**Pull request closed without merging**